zarr/storage.py

```diff
@@ -128,7 +128,7 @@ def normalize_store_arg(store: Any, clobber=False, storage_options=None, mode="w
         return store
 
 
-def rmdir(store: StoreLike, path: Path = None):
+def rmdir(store: StoreLike, path: str = None):
```

Comment:
Might need some comparison to #768

Reply:
I had a look at #768:
* I understand it would be nice if `path` could be `PathLike` in addition to `str`, wouldn't it?
* BTW, I would rename `_might_close()` to `_is_path_like()`, it's easier to understand because that's really what the function does.